packages/apollo-server-codegen/src/IR/index.ts

```diff
@@ -0,0 +1,75 @@
+import { DocumentNode } from "graphql";
+import { Description } from "./Descriptions";
+import { EnumDefinition } from "./Enums";
+import {
+  FieldDefinition,
+  ObjectDefinition,
+  TypelessObjectDefinition
+} from "./Objects";
+import { ArgumentDefinition, ResolverDefinition } from "./Resolvers";
+import { ScalarDefinition } from "./Scalars";
+import { CompoundType, ListType, NamedType, NonNullType } from "./Types";
+
+export {
+  ResolverDefinition,
+  ArgumentDefinition,
+  Description,
+  FieldDefinition,
+  ObjectDefinition,
+  CompoundType,
+  ListType,
+  NamedType,
+  NonNullType,
+  EnumDefinition,
+  ScalarDefinition
+};
+
+export function sdlToIR(
+  sdl: DocumentNode
+): [ObjectDefinition[], EnumDefinition[], ScalarDefinition[]] {
+  const objectDefinitions: TypelessObjectDefinition[] = [];
+  const enumDefinitions = [];
+  const scalarDefinitions = [];
+  for (const definition of sdl.definitions) {
+    switch (definition.kind) {
+      case "ObjectTypeDefinition":
+      case "ObjectTypeExtension":
+        objectDefinitions.push(new TypelessObjectDefinition(definition));
+        break;
+      case "ScalarTypeDefinition":
+        scalarDefinitions.push(new ScalarDefinition(definition));
+        break;
+      case "EnumTypeDefinition":
+        enumDefinitions.push(new EnumDefinition(definition));
+        break;
+      default:
+        console.log(
+          `Ignoring value of type ${definition.kind}: ${JSON.stringify(
+            definition
+          )}`
+        );
+    }
+  }
+
+  const flatMap = <T, E>(
```

Comment:
You may consider importing `apollo-env` (which polyfills `Array.flatMap()`)

Reply:
Saw that, but decided to not import the whole thing just for a polyfill if that's good by you.

Reply:
Yep, fine by me 👍

Reply:
Actually went back and decided to just go for the import lol